pkg/proxy/conntrack/fake.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conntrack
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// FakeInterface implements Interface by just recording entries that have been cleared.
```

Comment:
You mean "`FakeInterface` doesn't allow testing how callers will behave when a call fails"? That's true, but `CleanStaleEntries` eats all returned errors anyway. (Also, if a caller wanted to test that they could write their own implementation of `Interface`; the method names are all exported.)

Longer term, it would be good to have a proper reconciler here, and then we can get rid of the `TODO` comments in `conntrack.go` about how stale entries will stick around until they time out if deletion fails.

Reply:
yeah, this is ok